protocol/src/test/java/org/fido/iot/protocol/OnDieTest.java

```diff
@@ -0,0 +1,180 @@
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.ByteArrayInputStream;
+import java.net.MalformedURLException;
+import java.nio.CharBuffer;
+import java.nio.file.Path;
+import java.security.PublicKey;
+import java.security.cert.CertPath;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+class OnDieTest {
+
+  OnDieCache onDieCache = null;
+
+  String b64TestSignature =
+          "ARDiywa9EaMjQZ0dNWO4CbxGEL0vujai1k2rk5D/baL+8xwBsQ4ZF/eL0V/yxtaafl11BJZ7rjnesm"
+        + "/H8i6Hq3r8DeObqqGDo88mVnibvb9z3zlYlLahzLkwkhxsoTRRzXIQ6km2Dm6hQX5zmRkUDiFtzadw"
+        + "MDfh+dPVQMlf/vNG1j5K";
+  String serialNo = "daltest";
+  String b64DeviceCert =
+          "MIIBszCCATqgAwIBAgIQcYhLQDPbPylyGiZ0lFRLwzAKBggqhkjOPQQDAzAeMRwwGgYDVQQDDBNDU0"
+        + "1FIFRHTCBEQUxfSTAxU0RFMB4XDTE5MDEwMTAwMDAwMFoXDTQ5MTIzMTIzNTk1OVowFzEVMBMGA1UE"
+        + "AwwMREFMIEludGVsIFRBMHYwEAYHKoZIzj0CAQYFK4EEACIDYgAE044GJ2MiK44UHXubptTvkGefiy"
+        + "rKO9ofn5v1yBVJcwpbYYTBjop/W01f7Gv7se7sMin8D1zfoEIQuahlijcsVWlG0CcB6LodLkxQi+IS"
+        + "D8MNbObYIt8EGIacVVOgPdSho0QwQjAfBgNVHSMEGDAWgBSuPjAqQWKsFmeOf7U8OWyMbE+tfTAPBg"
+        + "NVHRMBAf8EBTADAQEAMA4GA1UdDwEB/wQEAwIDyDAKBggqhkjOPQQDAwNnADBkAjAdss2kczBguN6s"
+        + "iidupV+ipN8bCVAYe3eZV7c3i9rhTpHipVdII1/ppdswzl2IXQ0CMHNeOFuvHe64S9m2JRbBXUSdJ7"
+        + "iNQwp/4+OdQUmWYs2mB7KqZpmDPGQkq5mDuygBaA==";
+
+
+  OnDieCache getTestOnDieCache() throws Exception {
+    if (onDieCache == null) {
+      onDieCache = new OnDieCache(
+              getClass().getClassLoader().getResource("cachedir").getFile(),
+              false,
+              null,
+              null);
+    }
+    return onDieCache;
+  }
+
+
+  PublicKey getPublicKey() throws CertificateException {
+    // get public key from cert path
+    byte [] certBytes = Base64.getDecoder().decode(b64DeviceCert);
+    CertificateFactory certificateFactory =
+            CertificateFactory.getInstance("X.509"); //TODO, BouncyCastleLoader.load());
+    Certificate cert = certificateFactory.generateCertificate(
+            new ByteArrayInputStream(certBytes));
+    return cert.getPublicKey();
+  }
+
+  // The following test is left disabled since it can fail depending on
+  // the test environment. The test requires access to the source URLs for
+  // OnDie certs and CRLs. If this test is run on a network with access then
+  // it will fail. We cannot guarantee that this will be true for all environments.
+  // Comment out the @Disabled if you want to run the test.
+  @Test
+  @DisplayName("OnDie cache download test")
+  @Disabled
+  void testOnDieCacheDownload(@TempDir Path tempDir) throws Exception {
+
+    try {
+      String sourceList = "https://tsci.intel.com/content/OnDieCA/certs/"
+        + "https://tsci.intel.com/content/OnDieCA/crls/";
```

Comment:
Can we define these URLs as constants since we are using them at multiple places

Reply:
will update

Reply:
done